src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,216 @@
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST landNode = getExprChildNode(ast, TokenTypes.LAND);
+        if (landNode == null) {
+            if (checkOuterNotEqual(ast) || checkOuterInstanceOf(ast)) {
+                log(ast, MSG_KEY);
+            }
+        }
+        else {
+            final DetailAST notEqualNode = landNode.findFirstToken(TokenTypes.NOT_EQUAL);
+            final DetailAST instanceOfNode =
+                    landNode.findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+            if (instanceOfNode != null && isNullTest(notEqualNode)
+                    && isTextEqual(instanceOfNode, notEqualNode)) {
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * To get {@link TokenTypes#LITERAL_INSTANCEOF} Node or
+     * {@link TokenTypes#NOT_EQUAL}'s full ident text.
+     *
+     * @param ast DetailAST
+     * @return string ident node's text
+     */
+    private static String getFullText(DetailAST ast) {
+        final DetailAST dotNode = ast.findFirstToken(TokenTypes.DOT);
+        final String fullText;
+        if (dotNode == null) {
+            fullText = ast.findFirstToken(TokenTypes.IDENT).getText();
+        }
+        else {
+            final FullIdent fullIdent = FullIdent.createFullIdent(dotNode);
+            fullText = fullIdent.getText();
+        }
+        return fullText;
+    }
+
+    private static boolean checkOuterNotEqual(DetailAST ast) {
+        final DetailAST outerNotEqualNode = getExprChildNode(ast, TokenTypes.NOT_EQUAL);
+        boolean error = false;
+        if (isNullTest(outerNotEqualNode)) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST instanceOfNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.LITERAL_INSTANCEOF);
+                if (instanceOfNode != null) {
+                    error = isTextEqual(instanceOfNode, outerNotEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    private static boolean checkOuterInstanceOf(DetailAST ast) {
+        final DetailAST outerInstanceOfNode =
+                getExprChildNode(ast, TokenTypes.LITERAL_INSTANCEOF);
+        boolean error = false;
+        if (outerInstanceOfNode != null) {
+            final DetailAST nestedIfNode = getNestedIfNode(ast);
+            if (nestedIfNode != null) {
+                final DetailAST notEqualNode =
+                        getExprChildNode(nestedIfNode, TokenTypes.NOT_EQUAL);
+                if (isNullTest(notEqualNode)) {
+                    error = isTextEqual(outerInstanceOfNode, notEqualNode);
+                }
+            }
+        }
+        return error;
+    }
+
+    /**
+     * To get specific Node that is child of {@link TokenTypes#EXPR} Node.
+     *
+     * @param ast DetailAST
+     * @param token int
```

Comment:
add more description to parameter

Reply:
you can refer to `{@link TokenTypes}` class

Reply:
done

Reply:
done